web/src/components/netflow-topology/element-panel.tsx

```diff
@@ -0,0 +1,358 @@
+import {
+  Chart,
+  ChartArea,
+  ChartScatter,
+  ChartAxis,
+  ChartGroup,
+  ChartVoronoiContainer,
+  ChartThemeColor
+} from '@patternfly/react-charts';
+import {
+  DrawerActions,
+  DrawerCloseButton,
+  DrawerHead,
+  DrawerPanelBody,
+  DrawerPanelContent,
+  Flex,
+  FlexItem,
+  Text,
+  TextContent,
+  TextVariants
+} from '@patternfly/react-core';
+import { BaseEdge, BaseNode, GraphElement } from '@patternfly/react-topology';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { MetricFunction, MetricType } from 'src/model/flow-query';
+import { TopologyMetrics } from '../../api/loki';
+import { humanFileSize } from '../../utils/bytes';
+import { roundTwoDigits } from '../../utils/count';
+import { getDateFromUnixString, twentyFourHourTime } from '../../utils/datetime';
+import './element-panel.css';
+
+export const ElementPanelContent: React.FC<{
+  element: GraphElement;
+  metrics: TopologyMetrics[];
+  metricFunction: MetricFunction;
+  metricType: MetricType;
+}> = ({ element, metrics, metricFunction, metricType }) => {
+  const { t } = useTranslation('plugin__network-observability-plugin');
+  const data = element.getData();
+  const type = element.getType();
+
+  const resourceInfos = React.useCallback(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (d: any) => {
+      let infos: React.ReactElement | undefined;
+      if (d.type && d.name) {
+        infos = (
+          <TextContent id="resourcelink" className="element-text-container grouped">
+            <Text component={TextVariants.h4}>{d.type}</Text>
+            <ResourceLink inline={true} kind={d.type} name={d.name} namespace={d.namespace} />
+          </TextContent>
+        );
+      }
+      if (d.addr) {
+        infos = (
+          <>
+            {infos}
+            <TextContent id="address" className="element-text-container grouped">
+              <Text component={TextVariants.h4}>{t('Address')}</Text>
+              <Text id="addressValue">{d.addr}</Text>
+            </TextContent>
+          </>
+        );
+      }
+      return infos;
+    },
+    [t]
+  );
+
+  const metricTitle = React.useCallback(() => {
+    if (metricFunction === 'rate') {
+      return t('Flows rate');
+    } else {
+      switch (metricFunction) {
+        case 'avg':
+          return t('Average {{type}} (1m frame)', { type: metricType });
+        case 'max':
+          return t('Max {{type}} (1m frame)', { type: metricType });
+        case 'sum':
+          return t('Total {{type}}', { type: metricType });
+        default:
+          return '';
+      }
+    }
+  }, [metricFunction, metricType, t]);
+
+  const metricValue = React.useCallback(
+    (v: number) => {
+      return metricFunction !== 'rate' && metricType === 'bytes' ? humanFileSize(v, true, 0) : roundTwoDigits(v);
+    },
+    [metricFunction, metricType]
+  );
+
+  const metricCounts = React.useCallback(
```

Comment:
I think this kind of pattern, callbacks returning JSX, could often be turned into a separate react component. It would help keep the parent component size more reasonable..

I'm not asking to refactor that right now, but if we could agree on what's a good sizing of components to have better code readability  that would be great :) (we don't necessarily want tons of super small components either, it's a right balance to find)

Reply:
Agreed, splitting in `React.FC` also allow to test them separately in jest

As soon as it can be shared in the same file it's OK for me